ktor-client/ktor-client-core/common/src/io/ktor/client/features/DefaultResponseValidation.kt

```diff
@@ -30,11 +30,12 @@ public fun HttpClientConfig<*>.addDefaultResponseValidation() {
             }
 
             val exceptionResponse = exceptionCall.response
+            val exceptionResponseText = exceptionResponse.readText()
```

Comment:
What if the content is not properly encoded?

Reply:
We also need a test with binary garbage content to ensure it doesn't lead to crash here

Reply:
That's what I mean